src/mxarray.jl

```diff
@@ -686,11 +686,10 @@ end
 
 # deep conversion from MATLAB variable to Julia array
 
-jvalue(mx::MxArray, ty::Type{Array})  = jarray(mx)
-jvalue(mx::MxArray, ty::Type{Vector}) = jvector(mx)
-jvalue(mx::MxArray, ty::Type{Matrix}) = jmatrix(mx)
-jvalue(mx::MxArray, ty::Type{Number}) = jscalar(mx)::Number
-jvalue(mx::MxArray, ty::Type{String}) = jstring(mx)::String
-jvalue(mx::MxArray, ty::Type{Dict}) = jdict(mx)
-jvalue(mx::MxArray, ty::Type{SparseMatrixCSC}) = jsparse(mx)
-
+convert(::Type{Array}, mx::MxArray)  = jarray(mx)
+convert(::Type{Vector}, mx::MxArray) = jvector(mx)
+convert(::Type{Matrix}, mx::MxArray) = jmatrix(mx)
+convert(::Type{Number}, mx::MxArray) = jscalar(mx)
+convert(::Type{String}, mx::MxArray) = jstring(mx)
+convert(::Type{Dict}, mx::MxArray) = jdict(mx)
+convert(::Type{SparseMatrixCSC}, mx::MxArray) = jsparse(mx)
```

Comment:
The type you are converting to are too different from the type you are converting from and the operation is almost certainly something that the user should request explicitly. E.g. this is not matlab and you shouldn't construct a 1x1 array, pass it to matlab and later implicitly use the matlab array as a julia scalar.

These can be defined as constructor, which are basically "explicit" convert.